[Purge] Remove inactive tool: dandelion

0xBora opened the pull request:
## Summary

This PR removes the builder tool **Dandelion APIs** from the Developer Portal.

## Rationale

- Confirmed with a former member Roberto (@rcmorano) that the public hosted endpoints are down and there are no current plans to restore them. The documentation site is outdated/not functional fully, and current efforts, if any, are not publicly accessible. Therefore, the tool is no longer usable in its listed form.

- @zxpectre might have more insights into an updated version of dandelion as there seems to be a [dandelion lite node](https://github.com/GameChangerFinance/dandelion-lite) being worked on in development(?). 

> Related Issue #1485

rphair commented:
this build failure emailed to the group (https://github.com/cardano-foundation/developer-portal/actions/runs/15888665722) was likely the result of concurrent merge conflicts resolved through the GitHub UI... running it again shows no actual build problem.

0xBora commented:
I now removed the documentation page at /docs/get-started/dandelion-apis.md, updated the sidebar navigation remove dandelion mentioned in overview.md, and removed all associated images. This should complete the full removal of the dandelion tool from the portal now.
